tasks/storage_deal.go

```diff
@@ -321,3 +324,15 @@ func logStages(info api.DealInfo, log LogStatus) {
 		)
 	}
 }
+
+func (sp *_StorageTask__Prototype) Of(miner string, maxPrice, size, startOffset int64, fastRetrieval, verified bool) StorageTask {
+	st := _StorageTask{
```

Comment:
if you wanted to go "proper ipld-prime", this should use an assembler, which will do extra checks like complain if there are non-optional fields which weren't set.

And the verbosity shouldn't be too bad if you use https://pkg.go.dev/github.com/ipld/go-ipld-prime/fluent/qp#example-package.

(same applies to all the other `_T{...}` expressions)